core/src/main/scala/cats/InvariantSemigroupal.scala

```diff
@@ -0,0 +1,28 @@
+package cats
+
+import simulacrum.typeclass
+
+/**
+  * [[InvariantSemigroupal]] is nothing more than something both invariant
+  * and Semigroupal. It comes up enough to be useful, and composes well
+  */
+@typeclass trait InvariantSemigroupal[F[_]] extends Semigroupal[F] with Invariant[F] { self =>
+
+   def composeApply[G[_]: Apply]: InvariantSemigroupal[λ[α => F[G[α]]]] =
```

Comment:
Any plan to test cover this one as well?

Reply:
Has this been addressed?

Reply:
Yep!